srt/src/async_lib.rs

```diff
@@ -224,7 +229,9 @@ impl AsyncWrite for AsyncStream {
             }
         }
         let sock = self.socket.raw();
-        match unsafe { sys::srt_send(sock, src.as_ptr() as *const sys::char, src.len() as _) } {
+        let data_size = self.max_send_payload_size.min(src.len());
```

Comment:
A `0` value for `SRTO_PAYLOADSIZE` has a special meaning:

> When set to 0, there's no limit for a single sending call.

– https://github.com/Haivision/srt/blob/master/docs/API/API-socket-options.md#SRTO_PAYLOADSIZE

So maybe the logic should be something like...

```suggestion
        let data_size = if self.max_send_payload_size != 0 { self.max_send_payload_size.min(src.len()) } else { src.len() };
```

?